ci-operator/step-registry/ibmcloud/provision/dedicatedhost/ibmcloud-provision-dedicatedhost-commands.sh

```diff
@@ -0,0 +1,217 @@
+    return 1
+}
+
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4
+    local tmp family class dhg status
+    if [[ ${rg} = "Default" ]]; then
+        run_command "${IBMCLOUD_CLI} target --unset-resource-group"
+    else
+        run_command "${IBMCLOUD_CLI} target -g $rg"
+    fi
+    run_command "${IBMCLOUD_CLI} is target --gen 2"
+    tmp=$(${IBMCLOUD_CLI} is dedicated-host-profile $dProfile --output JSON | jq -r '.family +","+ .class')
+    family=${tmp%,*}
+    class=${tmp#*,}
+    echo "family:" $family " class:" $class
+    dhg="${dhName}g"
+    dhg=$(${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg} --output JSON | jq -r .id)
+    
+    ${IBMCLOUD_CLI} is dhg $dhg
+
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
+    
+    status=$(waitingStatus ${dhName})
+    echo "${dhName} status: ${status}"
+    run_command "${IBMCLOUD_CLI} is dedicated-host ${dhName}"
+
+    if [[ "${status}" != "stable" ]]; then
+        echo "ERROR: fail to create the dedicated host with profile $dProfile on $zone $rg"
+        deleteDedicatedHost "${dhName}" "${dhg}"
+    else
+        echo "${dhName}" > "${SHARED_DIR}/dedicated_host"
+        echo "${dhg}" > "${SHARED_DIR}/dedicated_host_group"
+    fi
+}
+
+function waitingStatus() {
+    local dh=$1 status
+    COUNTER=0
+    while [ $COUNTER -lt 200 ]
+    do 
+        sleep 10
+        COUNTER=`expr $COUNTER + 20`
+        status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+        if [[ "${status}" != "pending" ]]; then
+            break
+        fi
+    done
+    echo ${status}
+}
+
+function deleteDedicatedHost() {
+    local dh=$1 dhg=$2 status
+    status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+
+    run_command "${IBMCLOUD_CLI} is dhd ${dh} -f"
+    run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+}
+
+function getVMProfile() {
+    local profileHeader=$1
+    local all_profiles str1 str2
+    all_profiles=$(${IBMCLOUD_CLI} is instance-profiles -q | awk '(NR>1) {print $1}')
+    str1=${all_profiles##*$profileHeader}
+    str2=$(echo ${str1}|cut -d " " -f1)
+    echo ${profileHeader}${str2}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+ibmcloud_login
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+resource_group="${CLUSTER_NAME}-rg"
+create_resource_group ${resource_group}
+echo "${resource_group}" > "${SHARED_DIR}/ibmcloud_resource_group"
+
+#####################################
+#######create the dedicated host#####
+#####################################
+
+## Create the dedicated host
+echo "$(date -u --rfc-3339=seconds) - Creating the Dedicated Host..."
+
+dhName="${CLUSTER_NAME}-dh"
+dhProfile1=$(randomGetDHProfile)
+echo "random selected DHProfile1: $dhProfile1"
+
+dhGroup=$(randomSelectRG)
```

Comment:
The line `create_resource_group` already created a resource group, but on this line, decide to use it or not in a random way?

Reply:
dhGroup is the resource group of the host group, which can be "Default" or the "${CLUSTER_NAME}-rg"

Reply:
so if `dhGroup` select "Default", what `"${CLUSTER_NAME}-rg"` is used for?